File: src/main/java/com/_14ercooper/worldeditor/blockiterator/IteratorManager.java
package com._14ercooper.worldeditor.blockiterator;

import java.util.HashMap;
import java.util.Map;

public class IteratorManager {

    final Map<String, BlockIterator> iterators = new HashMap<>();

    // Register a new block iterator
    public void addIterator(String key, BlockIterator iterator) {
        iterators.put(key, iterator);
    }

    // Get the referenced block iterator
    public BlockIterator getIterator(String key) {
        return iterators.get(key);
    }
}

File: src/main/java/com/_14ercooper/worldeditor/blockiterator/IteratorManager.kt
// Licensed Under GPL v3.0
//https://github.com/14ercooper/14erEdit


package com._14ercooper.worldeditor.blockiterator

object IteratorManager {
    val iterators: MutableMap<String, BlockIterator> = HashMap()

    // Register a new block iterator
    fun addIterator(key: String, iterator: BlockIterator) {
        iterators[key] = iterator
    }

    // Get the referenced block iterator
    fun getIterator(key: String): BlockIterator? {
        return iterators[key]
    }
}